superset-frontend/spec/helpers/testing-library.tsx

```diff
@@ -62,14 +70,20 @@ export function createWrapper(options?: Options) {
     }
 
     if (useRedux) {
-      const mockStore =
-        store ??
-        configureStore({
-          preloadedState: initialState || {},
+      let storeWithInitialState;
+      if (initialState || reducers) {
+        storeWithInitialState = configureStore({
+          preloadedState: initialState,
           devTools: false,
-          reducer: reducers || reducerIndex,
+          reducer: {
+            ...(reducers || reducerIndex),
+            [api.reducerPath]: api.reducer,
+          },
+          middleware: getDefaultMiddleware =>
+            getDefaultMiddleware().concat(api.middleware),
         });
-
+      }
+      const mockStore = store ?? storeWithInitialState ?? storeWithApi;
```

Comment:
I think the logic here should consider the `store` parameter. There's no difference between `storeWithInitialState` and `storeWithApi` with the exception of the `initialState`. Both use the API and the reducers should be the same. I suggest something like:

`const mockStore = store ?? configureStore({...,  preloadedState: initialState || {}}) `